walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -14,28 +14,57 @@
 /** Helps persist and retrieve job posts. */
 public final class JobsDatabase {
     private static final String JOB_COLLECTION = "Jobs";
+    private static final String JOB_ID_FIELD = "jobId";
 
     /**
      * Adds a newly created job post.
      *
      * @param newJob Newly created job post. Assumes that it is non-nullable.
-     * @return Future for the added job post task, convenient for testing.
+     * @return A future of the detailed information of the writing.
      */
-    public Future<DocumentReference> addJob(Job newJob) {
-        return FireStoreUtils.getFireStore()
-                .collection(JOB_COLLECTION)
-                .add(newJob);
+    public Future<WriteResult> addJob(Job newJob) {
+        // Add document data after generating an id.
+        DocumentReference addedDocRef = FireStoreUtils.getFireStore()
+                .collection(JOB_COLLECTION).document();
+
+        String jobId = addedDocRef.getId();
+
+        // Creates a new Job object whose jobId is the autogenerated cloud firestore id
+        Job job = Job.newBuilder()
```

Comment:
wondering: why do you need to create another job object when you already have newJob? i suspect because of your json parsing meaning you might end up with a Job object that is not correct unless you re-run it with your builder. That's a definite red flag right here.